enterprise/frontend/src/embedding-sdk/store/auth.ts

```diff
@@ -0,0 +1,158 @@
+          api.sessionToken = session.id;
+        }
+      };
+      // verify that the session is actually valid before proceeding
+      await dispatch(getOrRefreshSession(sdkConfig.jwtProviderUri!)).unwrap();
+    } else if (isValidApiKeyConfig) {
+      // API key setup
+      api.apiKey = sdkConfig.apiKey;
+    }
+    // Fetch user and site settings
+    const [user, siteSettings] = await Promise.all([
+      dispatch(refreshCurrentUser()),
+      dispatch(refreshSiteSettings({})),
+    ]);
+
+    if (!user.payload) {
+      // The refresh user thunk just returns null if it fails to fetch the user, it doesn't throw
+      const error = new Error(
+        "Failed to fetch the user, is the session valid?",
+      );
+      console.error(error);
+      throw error;
+    }
+    if (!siteSettings.payload) {
+      const error = new Error(
+        "Failed to fetch the site settings, is the session valid?",
+      );
+      console.error(error);
+      throw error;
+    }
+  },
+);
+
+export const refreshTokenAsync = createAsyncThunk(
+  "sdk/token/REFRESH_TOKEN",
+  async (url: string, { getState }): Promise<EmbeddingSessionToken | null> => {
+    // The SDK user can provide a custom function to refresh the token.
+    const customGetRefreshToken = getFetchRefreshTokenFn(
+      getState() as SdkStoreState,
+    );
+
+    const getRefreshToken = customGetRefreshToken ?? defaultGetRefreshTokenFn;
+
+    // # How does the error handling work?
+    // This is an async thunk, thunks _by design_ can fail and no error will be shown on the console (it's the reducer that should handle the reject action)
+    // The following lines are wrapped in a try/catch block that will catch any errors thrown, log them to the console as a big red errors, and re-throw them to make the thunk reject
+    // In this way we also support standard thrown Errors in the custom fetchRequestToken user provided function
+
+    try {
+      const response = await getRefreshToken(url);
+      const source = customGetRefreshToken
+        ? '"fetchRequestToken"'
+        : "jwtProvider endpoint";
+
+      if (!response || typeof response !== "object") {
+        throw new Error(
+          `The ${source} must return an object with the shape {id:string, exp:number, iat:number, status:string}, got ${safeStringify(response)} instead`,
+        );
+      }
+      if ("status" in response && response.status !== "ok") {
+        if ("message" in response && typeof response.message === "string") {
+          // For some errors, the BE gives us a message that explains it
+          throw new Error(response.message);
+        }
+        if ("status" in response && typeof response.status === "string") {
+          // other times it just returns an error code
+          throw new Error(
+            `Failed to refresh token, got status: ${response.status}`,
+          );
+        }
+      }
+      // Lastly if we don't have an error message or status check if we actually got the session id
```

Comment:
```suggestion
      // Lastly if we don't have an error message or status, check if we actually got the session ID
```

Let me know if I interpret this right. Without a comma, it was confusing at first. As I read "status check" as a single word.

Reply:
Correct, thanks for the correction 🙏